File: frontend/src/types/analysis.ts
export interface AnalysisResult {
  id: string;
  meeting_id: string;
  clarity_score: number;
  commitment_score: number;
  contradiction_count: number;
  vague_statement_count: number;
  unresolved_count: number;
  summary: {
    overview?: string;
    key_themes?: string[];
  };
  decisions: string[];
  suggested_followups: string[];
  created_at?: string;
}

export interface Flag {
  id: string;
  meeting_id: string;
  segment_id?: string;
  flag_type: "vague" | "contradiction" | "missing_owner" | "missing_deadline";
  severity: "low" | "medium" | "high";
  explanation: string;
  evidence: Record<string, unknown>;
  user_verdict?: "accepted" | "rejected" | null;
  created_at?: string;
}

export interface ActionItem {
  id: string;
  meeting_id: string;
  segment_id?: string;
  description: string;
  owner: string;
  deadline?: string;
  priority: "low" | "medium" | "high";
  status: "identified" | "confirmed" | "completed";
  dependency?: string;
  created_at?: string;
}

export interface TranscriptSegment {
  id: string;
  meeting_id: string;
  segment_index: number;
  speaker_label: string;
  text: string;
  start_time?: number;
  end_time?: number;
}

File: backend/app/schemas/analysis.py
"""
Pydantic schemas for Analysis API responses.
"""
from pydantic import BaseModel
from typing import Optional, List, Any


class FlagResponse(BaseModel):
    id: str
    meeting_id: str
    segment_id: Optional[str] = None
    flag_type: str
    severity: str
    explanation: str
    evidence: Optional[Any] = None
    user_verdict: Optional[str] = None
    created_at: Optional[str] = None


class ActionItemResponse(BaseModel):
    id: str
    meeting_id: str
    segment_id: Optional[str] = None
    description: str
    owner: Optional[str] = None
    deadline: Optional[str] = None
    priority: Optional[str] = None
    status: Optional[str] = None
    dependency: Optional[str] = None
    created_at: Optional[str] = None


class AnalysisResponse(BaseModel):
    id: str
    meeting_id: str
    clarity_score: int
    commitment_score: Optional[int] = None
    contradiction_count: int
    vague_statement_count: int
    unresolved_count: int
    summary: Optional[Any] = None
    decisions: Optional[Any] = None
    suggested_followups: Optional[Any] = None
    created_at: Optional[str] = None
